ceph/ceph_admin/orch.py

```diff
@@ -237,3 +242,143 @@ def op(self, op, config):
+        """
+        out, _ = self.shell(["ceph", "orch", "status", "-f", "json"])
+        orch_paused = loads(out)["paused"]
+
+        if op == "pause" and orch_paused:
+            LOG.info("The orch operations are paused")
+            return True
+        elif op == "resume" and not orch_paused:
+            LOG.info("The orch operations are resumed")
+            return True
+
+        return False
+
+    def check_host_status(self, op, node) -> bool:
+        """
+        Check ceph orchestrator status to verify pause and resume orch operations using ceph orch status command
+        Args:
+            op: operation after which host status is checked. ex: enter|exit
+            node: CephNode for which status needs to be checked.
+
+        Returns:
+            true if orchestrator operation has been performed successfully, else false.
+        """
+        out, _ = self.shell(["ceph", "orch", "host", "ls", "-f", "json"])
+        status = [h["status"] for h in loads(out) if h["hostname"] == node.hostname][0]
+        retry_count = 3
+
+        if op == "enter" and status == "maintenance":
+            LOG.info(f"The node {node.hostname} has entered maintenance mode")
+            count = 0
+            active_daemon = True
+            while count < retry_count:
+                sleep(20)
+                out, _ = self.shell(["ceph", "orch", "ps", node.hostname, "-f", "json"])
+                daemon_status = loads(out)
+                active_daemon = [
+                    daemon for daemon in daemon_status if daemon["is_active"]
+                ]
+                if active_daemon:
+                    count += 1
+                else:
+                    break
+            # add verification for ceph health status to see if the daemons corresponding to this node have gone down
+            return not bool(active_daemon)
+        elif op == "exit" and status != "maintenance":
+            LOG.info(f"The node {node.hostname} is not in maintenance mode")
+            count = 0
+            inactive_daemon = True
+            while count < retry_count:
+                sleep(20)
+                out, _ = self.shell(["ceph", "orch", "ps", node.hostname, "-f", "json"])
+                daemon_status = loads(out)
+                inactive_daemon = [
+                    daemon for daemon in daemon_status if not daemon["is_active"]
+                ]
+                if inactive_daemon:
+                    count += 1
+                else:
+                    break
+            # add verification for ceph health status to see if the cluster is back to healthy state wrt this node
+            return not bool(inactive_daemon)
+
+        return False
+
+    def get_active_mgr(self) -> str:
+        """
+        Fetch the current active mgr node of the cluster.
+        Returns:
+            active mgr hostname
+        """
+        out, _ = self.shell(args=["ceph", "-s"])
```

Comment:
Yes, but the json output does not have the information of the node name in which mgr is deployed (I verified this by executing the command with json format). So since I need the mgr node details here I retained it without json formatting.